spec/spec-compat.md

```diff
@@ -0,0 +1,139 @@
+
+# Wasm Image Specification v0.0.0
+
+## Introduction:
+
+This document describes a varient of [Wasm Artifact Image Specification](spec.md), which leverages the compatible layer media types. Here, we omit definition and terminology explained in [Wasm Artifact Image Specification](spec.md). 
+
+We call this variant "compat", and the spec in [Wasm Artifact Image Specification](spec.md) "oci".
+
+## Description
+
+This *compat* variant makes use of compatible media type for layers, and is not based on custom OCI Artifact media types. This way users can oeperate with standard tools such as docker, podman, buildah, etc.
+
+## Specification
+
+### Layer
+
+The *compat* variant must consist of exactly one layer whose media type is one of the followings:
+- `application/vnd.oci.image.layer.v1.tar+gzip`
+- `application/vnd.docker.image.rootfs.diff.tar.gzip`
+
+In addition, the layer must consist of the following two files:
+- `plugin.wasm` - (**Required**) A Wasm binary to be loaded by the runtime.
+- `runtime-config.json` - (**Optional**) A runtime configuration specified in [Wasm Artifact Image Specification](spec.md#Format). This is mainly used as metadata for the image, depending on the runtime.
+
+### Annotation
+
+If the media type equals `application/vnd.oci.image.layer.v1.tar+gzip`, then a *compat* variant image must add the annotation `module.wasm.image/variant=compat` in the manifest.
```

Comment:
Why only for this media type? If it is not easily added by the docker cli, maybe we can just relax this requirement to a "should"

Reply:
OK I changed the implementation so that we remove the need for annotations: https://github.com/istio/istio/commit/9975a786bb104088f5561223c320f8b507378ffa

Will change the text.

Reply:
https://github.com/solo-io/wasm/pull/265/commits/45316b2bdafe98962c87ac675f46499f59722d16 done.